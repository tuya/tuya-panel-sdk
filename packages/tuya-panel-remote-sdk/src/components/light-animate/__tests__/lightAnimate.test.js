/**
 * @jest-environment jsdom
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import LightAnimate from '../index';

// jest.mock('react-native-svg', () => {
//   const realRectSvg = jest.requireActual('react-native-svg');
//   return {
//     ...realRectSvg,
//     Rect: ({ children }) => <>{children}</>,
//     Circle: ({ children }) => <>{children}</>,
//   };
// });
// jest.mock('react-native', () => {
//   const rn = jest.requireActual('react-native');
//   return {
//     ...rn,
//     Image: () => null,
//   };
// });

describe('LightAnimate components', () => {
  it('basic render', () => {
    const wrapper = shallow(<LightAnimate type={null} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('lightTempMinus render', () => {
    let type = null;
    const wrapper = shallow(<LightAnimate type={type} />);
    type = 'lightTempMinus';
    wrapper.setProps({ type });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('lightTempAdd render', () => {
    let type = 'lightTempMinus';
    const wrapper = shallow(<LightAnimate type={type} />);
    type = 'lightTempAdd';
    wrapper.setProps({ type });
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('lightBrightAdd render', () => {
    const duration = 2000;
    const wrapper = shallow(<LightAnimate type={null} duration={duration} />);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
