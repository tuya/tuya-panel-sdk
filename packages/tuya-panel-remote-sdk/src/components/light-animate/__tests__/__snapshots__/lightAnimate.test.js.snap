// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightAnimate components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Object {
        "height": 480,
        "opacity": 0,
        "width": 750,
      }
    }
  >
    <RadialGradient
      cx="50%"
      cy="50%"
      fx="50%"
      fy="50%"
      gradientId="Gradient"
      rx="50%"
      ry="50%"
      stops={
        Array [
          Object {
            "offset": "0%",
            "stopColor": "#FFFFFF",
            "stopOpacity": "1",
          },
          Object {
            "offset": "100%",
            "stopColor": "#FFFFFF",
            "stopOpacity": "0",
          },
        ]
      }
      style={
        Object {
          "height": 480,
          "width": 750,
        }
      }
    />
  </AnimatedComponent>
  <Image
    source={null}
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 60,
        },
        null,
      ]
    }
  />
</View>
`;

exports[`LightAnimate components lightBrightAdd render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Object {
        "height": 480,
        "opacity": 0,
        "width": 750,
      }
    }
  >
    <RadialGradient
      cx="50%"
      cy="50%"
      fx="50%"
      fy="50%"
      gradientId="Gradient"
      rx="50%"
      ry="50%"
      stops={
        Array [
          Object {
            "offset": "0%",
            "stopColor": "#FFFFFF",
            "stopOpacity": "1",
          },
          Object {
            "offset": "100%",
            "stopColor": "#FFFFFF",
            "stopOpacity": "0",
          },
        ]
      }
      style={
        Object {
          "height": 480,
          "width": 750,
        }
      }
    />
  </AnimatedComponent>
  <Image
    source={null}
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 60,
        },
        null,
      ]
    }
  />
</View>
`;

exports[`LightAnimate components lightTempAdd render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Object {
        "height": 480,
        "opacity": 0,
        "width": 750,
      }
    }
  >
    <RadialGradient
      cx="50%"
      cy="50%"
      fx="50%"
      fy="50%"
      gradientId="Gradient"
      rx="50%"
      ry="50%"
      stops={
        Array [
          Object {
            "offset": "0%",
            "stopColor": "#F3E7A6",
            "stopOpacity": "1",
          },
          Object {
            "offset": "100%",
            "stopColor": "#F3E7A6",
            "stopOpacity": "0",
          },
        ]
      }
      style={
        Object {
          "height": 480,
          "width": 750,
        }
      }
    />
  </AnimatedComponent>
  <Image
    source={null}
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 60,
        },
        null,
      ]
    }
  />
</View>
`;

exports[`LightAnimate components lightTempMinus render 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "width": "100%",
      },
      null,
    ]
  }
>
  <AnimatedComponent
    style={
      Object {
        "height": 480,
        "opacity": 0,
        "width": 750,
      }
    }
  >
    <RadialGradient
      cx="50%"
      cy="50%"
      fx="50%"
      fy="50%"
      gradientId="Gradient"
      rx="50%"
      ry="50%"
      stops={
        Array [
          Object {
            "offset": "0%",
            "stopColor": "#9BCCFF",
            "stopOpacity": "1",
          },
          Object {
            "offset": "100%",
            "stopColor": "#9BCCFF",
            "stopOpacity": "0",
          },
        ]
      }
      style={
        Object {
          "height": 480,
          "width": 750,
        }
      }
    />
  </AnimatedComponent>
  <Image
    source={null}
    style={
      Array [
        Object {
          "position": "absolute",
          "top": 60,
        },
        null,
      ]
    }
  />
</View>
`;
